sycl/include/sycl/ext/intel/experimental/fpga_annotated_properties.hpp

```diff
@@ -31,54 +31,68 @@ using property_value =
 //===----------------------------------------------------------------------===//
 //        FPGA properties of annotated_arg/annotated_ptr
 //===----------------------------------------------------------------------===//
-struct register_map_key {
+struct register_map_key
+    : oneapi::experimental::detail::compile_time_property_key<
+          oneapi::experimental::detail::PropKind::RegisterMap> {
   using value_t = property_value<register_map_key>;
 };
 
-struct conduit_key {
+struct conduit_key : oneapi::experimental::detail::compile_time_property_key<
+                         oneapi::experimental::detail::PropKind::Conduit> {
   using value_t = property_value<conduit_key>;
 };
 
-struct stable_key {
+struct stable_key : oneapi::experimental::detail::compile_time_property_key<
+                        oneapi::experimental::detail::PropKind::Stable> {
   using value_t = property_value<stable_key>;
 };
 
-struct buffer_location_key {
+struct buffer_location_key
+    : oneapi::experimental::detail::compile_time_property_key<
+          oneapi::experimental::detail::PropKind::BufferLocation> {
```

Comment:
Would that be always ABI compatible taking into account that some properties are used in sycl/source? E.g.: https://github.com/intel/llvm/blob/sycl/sycl/source/detail/usm/usm_impl.cpp#L98-L99

Reply:
The ones in that example are from SYCL 2020 properties, while these properties are all from the [sycl_ext_oneapi_properties](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_properties.asciidoc) extension. Since that property-list is templated I don't believe it crosses the library boundaries, so it would only be if we pass the properties around by themselves.